misc/bazel/internal/install.py

```diff
@@ -42,8 +44,20 @@
 script = runfiles.Rlocation(opts.pkg_install_script)
 
 if destdir.exists() and opts.cleanup:
-    shutil.rmtree(destdir)
-
+    if platform.system() == 'Windows':
+        # On Windows we might have virus scanner still looking at the path so
+        # attempt removal a couple of times sleeping between each attempt.
+        for attempt in [1, 2]:
```

Comment:
I think there's a bug here:
1. we enter the loop with `attempt == 1`.
1. we try `rmtree`. Let's suppose we fail with `winerror == 32`
2. we sleep 1s
3. we reenter the loop with `attempt == 2`.
4. we try `rmtree` again. Let's suppose we fail again with `winerror == 32`
5. we sleep for 2s, but then...
6. ...we exit the loop and we move on, having silently failed to remove the directory, and having slept 2s for no reason

we need to do a final `rmtree` after the `for` unless we succeeded. We could do that by adding a
```python
else:
   shutil.rmtree(destdir)
```
after the `for` loop, but I know not all are comfortable with the `for: ... else: ...` construct (`else` is visited only if the `for` was not interrupted by a `break`)

Reply:
Fixed. The code is _exceedingly_ ugly now though :(